game/src/main/org/apollo/game/message/impl/ChatMessage.java

```diff
@@ -49,7 +35,7 @@ public ChatMessage(String message, byte[] compressedMessage, int color, int effe
 	 *
 	 * @return The compressed message.
 	 */
-	public byte[] getCompressedMessage() {
+	public final byte[] getCompressedMessage() {
 		return compressedMessage;
```

Comment:
This needs to return `compressedMessage.clone()`. We clone it so that code that calls `getCompressedMessage()` can't mutate the message text (as arrays are inherently mutable), and all of the `Message` classes should be immutable.